src/Atmos/Model/AtmosModel.jl

```diff
@@ -0,0 +1,122 @@
+
+import CLIMA.DGmethods: BalanceLaw, dimension, vars_aux, vars_state, num_state_for_gradtransform,  vars_gradtransform, vars_diffusive,
+  flux!, source!, wavespeed, boundarycondition!, gradtransform!, diffusive!,
+  init_aux!, init_state!, preodefun!
+
+struct AtmosModel{T,M,R} <: BalanceLaw
+  turbulence::T
+  moisture::M
+  radiation::R
+end
+
+using CLIMA.PlanetParameters: R_d, cp_d, grav, cv_d, MSLP, T_0
+
+const γ_exact = 7 // 5
+
+vars_state(m::AtmosModel) = (:ρ, :ρu, :ρv, :ρw, :ρe, vars_state(m.moisture)...)
+num_state_for_gradtransform(::AtmosModel) = length(vars_state(m)) #TODO: need to handle this better
+
+vars_gradtransform(::AtmosModel) = (vars_gradtransform(m.turbulence)...,)
+vars_diffusive(::AtmosModel) = (vars_diffusive(m.turbulence)..., 
+                                vars_diffusive(m.moisture)..., 
+                                vars_diffusive(m.radiation)...)
+
+vars_aux(::AtmosModel) = (vars_aux(m.turbulence)...,
+                          vars_aux(m.moisture)...,
+                          vars_aux(m.radiation)...)
+
+# Navier-Stokes flux terms
+function flux!(m::AtmosModel, flux::Grad, state::State, diffusive::State, auxstate::State, t::Real)
+  # preflux
+  ρinv = 1 / state.ρ
+  u, v, w = ρinv * state.ρu, ρinv * state.ρv, ρinv * state.ρw
+
+  P = pressure(m.moisture, state, diffusive, auxstate, t)
```

Comment:
lowercase `p` for pressure is more common (we use uppercase `P` for precipitation).